core/models.py

```diff
@@ -165,3 +168,30 @@ def image_url(self: "Image") -> str:
             return self.image.url
         except ValueError:
             return ""
+
+
+class CommunityMember(models.Model):
+    ADMIN: ClassVar[str] = "admin"
+    MODERATOR: ClassVar[str] = "moderator"
+    MEMBER: ClassVar[str] = "member"
+
+    ROLE_CHOICES: ClassVar[list[tuple[str, str]]] = [
+        (ADMIN, "Admin"),
+        (MODERATOR, "Moderator"),
+        (MEMBER, "Member"),
+    ]
+
+    community = models.ForeignKey(
+        Community,
+        on_delete=models.CASCADE,
+        related_name="memberships",
+    )
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="communities")

```

Comment:
Are you sure about using on_delete cascade here? 
And on community?

Reply:
I think so - if we delete a user, how can he be a member of the community later, just as we delete the community, the memberships should also be deleted

Reply:
True, thanks, I got confused.